xarray/test/test_backends.py

```diff
@@ -492,6 +492,21 @@ def create_tmp_file(suffix='.nc', allow_cleanup_failure=False):
             if not allow_cleanup_failure:
                 raise
 
+@contextlib.contextmanager
+def create_tmp_files(nfiles, suffix='.nc', allow_cleanup_failure=False):
```

Comment:
Any reason why you can't reuse `create_tmp_file` internally here?

This would be most straightforwardly done with `contextlib.ExitStack`, though we would need a backport to Python 2.7:
https://docs.python.org/3.6/library/contextlib.html#contextlib.ExitStack